interpolate.cc

```diff
@@ -108,12 +110,14 @@ interpfilt_process(void)
             wpt_new->shortname = QString();
             wpt_new->description = QString();
 
+            frac = (double)(timen - time1) / (double)(wpt->creation_time.toTime_t() - time1);
```

Comment:
frac is just between 0 and 1 here and does not hold the time. You're right though, toTime_t() will break before 1970 and after 2106. Replaced it with toSecsSinceEpoch().